New reducers added (toBounds and toAmplitude)

Smoren opened the pull request:
1. Methods added:
  * `Reduce::toBounds()`,
  * `Reduce::toAmplitude()`,
  * `Stream::toBounds()`,
  * `Stream::toAmplitude()`.
2. All the new methods are covered by tests.
3. README updated with description of new methods.

Smoren committed:
1. Methods added:
  * `Reduce::toBounds()`,
  * `Reduce::toAmplitude()`,
  * `Stream::toBounds()`,
  * `Stream::toAmplitude()`.
2. All the new methods are covered by tests.
3. README updated with description of new methods.

Smoren commented:
I've rebased this branch from `develop`.

Smoren commented:
I've rebased this branch from `develop` again and resolved the conflicts.

markrogoyski commented on src/Reduce.php:
```diff
@@ -154,4 +154,36 @@ public static function toString(iterable $data, string $separator = '', string $
+     *
+     * @return array{numeric, numeric}|array{null, null}
+     */
+    public static function toBounds(iterable $numbers): array
```
Suggestion: Since `toMin` and `toMax` are already present, for consistency what do you think about naming this `toMinMax`. It also implies there are two values. "To bounds" on the other hand could be confused with restricting or adjusting something to be within some predefined bounds condition perhaps.

Smoren replied:
Done.

markrogoyski commented on src/Reduce.php:
```diff
@@ -154,4 +154,36 @@ public static function toString(iterable $data, string $separator = '', string $
+     *
+     * @return int|float
+     */
+    public static function toAmplitude(iterable $numbers)
```
Suggestion: Let's call this `toRange`. Range is the max - min. Amplitude is usually half the distance of the range. For a wave, it is from x = 0 to the peek.

See https://www.mathwords.com/a/amplitude.htm

Smoren replied:
Done.

markrogoyski commented:
Hi @Smoren,

Thanks for the PR for the new functionality. I left just a couple suggestions for clearer names of these two functions.

Thanks,
Mark

Smoren committed:
1. `Reduce::toBounds()`, `Stream::toBounds()` renamed to `toMinMax()`.
2. `Reduce::toAmplitude()`, `Stream::toAmplitude()` renamed to `toRange()`.
3. Tests and README updated.

Smoren commented:
@markrogoyski 

Hi Mark! Thank you for your suggestions! All of them are applied.

Summary:

1. `Reduce::toBounds()`, `Stream::toBounds()` renamed to `toMinMax()`.
2. `Reduce::toAmplitude()`, `Stream::toAmplitude()` renamed to `toRange()`.
3. Tests and README updated.

Smoren commented:
@markrogoyski 

BTW, what do you think, will there be confusion with the method name `getRange()`?

I mean, range is often understood as a generator (as in python: `range(start, size, step)`).

markrogoyski commented:
Hi @Smoren,

I'm not concerned. The English definition of range ([See noun 7](https://www.merriam-webster.com/dictionary/range)) as well as the [computing definition](https://www.techopedia.com/definition/4422/range) and general [math definition](https://www.mathsisfun.com/definitions/range-statistics-.html) and [statistics definition](https://en.wikipedia.org/wiki/Range_(statistics)) of range describe the difference between max and min. Also, all of the `toXXX` methods describe reducing to a non-stream value, so it is consistent with that.

Also, in Python, there are two interfaces for `range`. The one that only takes a single value actually takes something resembling what the "range" actually is from the definition (although slightly different):
```python
> list(range(10))
> [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```

Smoren commented:
Thank you for the answer and for merging my PR!
